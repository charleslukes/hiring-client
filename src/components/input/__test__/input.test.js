import React from "react";
import { shallow, configure } from "enzyme";
import Input from "../index";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

// React 16 Enzyme adapter
configure({ adapter: new Adapter() });

describe("<Input />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const onChange = jest.fn();

      const wrapper = shallow(
        <Input
          validate={{
            regexp: /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
            message:
              "Password must contain at least 8 characters, 1 letter, and 1 number"
          }}
          placeholder="dummyPlaceholder"
          value="dummyValue"
          name="dummyName"
          type="dummyType"
          onChange={onChange}
        />
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
