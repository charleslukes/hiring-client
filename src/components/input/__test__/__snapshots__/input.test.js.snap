// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> render() renders the component 1`] = `
<WithTheme(FormField)
  name="dummyName"
  onChange={[MockFunction]}
  placeholder="dummyPlaceholder"
  required={true}
  style={
    Object {
      "borderRadius": "20px",
      "marginBottom": "15px",
    }
  }
  type="dummyType"
  validate={
    Object {
      "message": "Password must contain at least 8 characters, 1 letter, and 1 number",
      "regexp": /\\^\\(\\?=\\.\\*\\[A-Za-z\\]\\)\\(\\?=\\.\\*\\\\d\\)\\[A-Za-z\\\\d\\]\\{8,\\}\\$/,
    }
  }
  value="dummyValue"
/>
`;
